src/BinDeps.jl

```diff
@@ -91,18 +91,27 @@ if is_unix()
         elseif extension == ".tar"
             return (`tar xf $file --directory=$directory`)
         elseif extension == ".zip"
-            @static if is_bsd() && !is_apple()
-                return (`unzip -x $file -d $directory $file`)
-            else
-                return (`unzip -x $file -d $directory`)
-            end
+            return (`unzip -x $file -d $directory`)
         elseif extension == ".gz"
             return pipeline(`mkdir $directory`, `cp $file $directory`, `gzip -d $directory/$file`)
         end
         error("I don't know how to unpack $file")
     end
 end
 
+if KERNEL == :FreeBSD
+    # The `tar` on FreeBSD can auto-detect the archive format via libarchive.
+    # The supported format can be found in libarchive-formats(5).
+    # For NetBSD and OpenBSD, libarchive is not available.
+    # For macOS, it does. But the previous unpack function works fine already.
+    function unpack_cmd(file, dir, ext, secondary_ext)
+        tar_args = ["--no-same-owner", "--no-same-permissions"]
```

Comment:
Yes, it's here: https://github.com/JuliaLang/BinDeps.jl/pull/267/files#diff-8bd91b8bf2fc0449a96b712e9150aad7R111 .

FYI, https://github.com/freebsd/freebsd-ports/blob/master/Mk/bsd.port.mk#L736-L738